fhir-registry/src/main/java/com/ibm/fhir/registry/spi/FHIRRegistryResourceProvider.java

```diff
@@ -8,44 +8,77 @@
 
 import java.util.Collection;
 
+import com.ibm.fhir.model.resource.Resource;
 import com.ibm.fhir.registry.resource.FHIRRegistryResource;
 
 /**
  * An SPI for {@link FHIRRegistryResource} instances
  */
 public interface FHIRRegistryResourceProvider {
     /**
-     * Get the FHIR registry resources for this provider. A FHIR registry resource contains all of the information necessary
-     * to load an actual FHIR resource into memory. FHIR registry resources must have a unique url+version pair. If this method
-     * returns a collection containing FHIR registry resources with duplicate url+version pairs, then the first one will be
-     * added to the registry and any subsequent duplicates will be ignored.
+     * Get the registry resource from this provider for the given resource type, url and version
      *
-     * @return
-     *     the FHIR registry resources for this provider.
-     */
-    Collection<FHIRRegistryResource> getResources();
-
-    /**
-     * Get the FHIR registry resource with the given url and version. If the version is null, then the latest version of
-     * the registry resource is returned.
+     * <p>If the version is null, then the latest version of the registry resource is returned (if available)
      *
+     * @param resourceType
+     *     the resource type of the registry resource
      * @param url
      *     the url of the registry resource
      * @param version
      *     the version of the registry resource
```

Comment:
worth mentioning that version is optional?  what is the behavior if its not present?

Reply:
i guess its right above this, so maybe not

Reply:
added to the javadoc